include/ttmlir/Dialect/TTKernel/IR/TTKernelOpsTypes.td

```diff
@@ -81,22 +81,24 @@ def TTKernel_CB : TTKernel_Type<"CB", "cb"> {
                           "uint64_t":$address,
                           "MemRefType":$memref,
                           "uint64_t":$page_size,
-                          "uint64_t":$num_buffers);
-    let assemblyFormat = "`<` $port`,` $address`,` $memref`,` $page_size`,` $num_buffers `>`";
+                          "uint64_t":$num_buffers,
+                          "bool":$is_internal);
+    let assemblyFormat = "`<` $port`,` $address`,` $memref`,` $page_size`,` $num_buffers`,` $is_internal `>`";
 
     let extraClassDeclaration = [{
         static CBType get(::mlir::MLIRContext *context,
                               CBPort port,
                               uint64_t address,
-                              MemRefType memref) {
+                              MemRefType memref,
+                              bool is_internal = false) {
           uint64_t numBuffers = 1;
           uint64_t pageSize = 0;
           if (::mlir::isa<::mlir::tt::TileType>(memref.getElementType())) {
             pageSize = ::mlir::cast<::mlir::tt::TileType>(memref.getElementType()).getSizeBytes();
           } else {
             pageSize = memref.getShape().back() * (memref.getElementType().getIntOrFloatBitWidth() / 8);
           }
-          return CBType::get(context, port, address, memref, pageSize, numBuffers);
+          return CBType::get(context, port, address, memref, pageSize, numBuffers, is_internal);
```

Comment:
Why do we need the `is_internal` designation?

Edit: I think this is the same question I have below

Reply:
Answerd in the other place.